File: python/completion.py
import json
from openai import OpenAI
# from dotenv import load_dotenv
from pprint import pprint
from utils.conversion import object_to_dict


load_dotenv()

client = OpenAI()

completion = client.chat.completions.create(
    model="gpt-4o",
    # n=1,
    messages=[
        {"role": "system", "content": "Отвечай кратко, но с объяснением."},
        # или второй вариант
        # {"role": "system", "content": "Отвечай кратко, но с объяснением.Отвечай в формате JSON"},
        
        {"role": "user", "content": "Сколько тебе лет?"},
    ],
    response_format={'type': 'text'},  
    # или второй вариант
    # response_format={'type': 'json_object'},
    
    # temperature = 0.9
    # или второй вариант
    top_p = 0.9,
    # stop = 'нет',
    
)

# Получаем текст из ответа
raw_content = completion.choices[0].message.content

# Преобразуем строку в JSON-объект (если это JSON)
try:
    parsed_json = json.loads(raw_content)
    print(json.dumps(parsed_json, indent=4, ensure_ascii=False))  # Красиво выводим JSON
except json.JSONDecodeError:
    print(raw_content)  # Если не JSON, просто печатаем ответ
    


pprint(object_to_dict(completion), indent = 1)

File: node/completion.mjs
import openai from "./utils/openai.mjs";
import { openaiErrorHandler } from "./utils/openaiErrorHandler.mjs";

const systemContent = "Старайся отвечать кратко. Не более 45 слов";
const userContent = "Напиши что нового у Google";

try {
  const completion = await openai.chat.completions.create({
    model: "gpt-4o-mini",
    max_tokens: 800,
    n: 2,
    messages: [
      {
        role: "system",
        content: systemContent,
      },
      {
        role: "user",
        content: userContent,
      },
    ],
  });

  // console.log(completion);

  completion.choices.forEach((choice) => {
    console.log(choice.message);
  });
} catch (error) {
  openaiErrorHandler(error);
}
